Add version info and description to cuberite CLI

Pandapip1 opened the pull request.

bearbin reviewed (commented):
Not sure if we should default to 'unknown' or 'none' for the version. Unknown sort of makes sense, but it looks like TCLAP uses 'none' and this wouldn't display the version at all - which seems more appropriate to me as it replicates the current behaviour.

Otherwise, looks good once somebody else with some more CMake expertise takes a look (and somebody also confirms that it works on Windows).

Finally, please review the LICENSE and add yourself to the correct position in the CONTRIBUTORS file to indicate that you wrote the code you're submitting and you license it under that license.

MightyFilipns commented:
Compiles and works on windows

Pandapip1 commented:
Reviewed the LICENSE; it seems that it hasn't been correctly applied, since the notice hasn't been completed.

```
   APPENDIX: How to apply the Apache License to your work.

      To apply the Apache License to your work, attach the following
      boilerplate notice, with the fields enclosed by brackets "[]"
      replaced with your own identifying information. (Don't include
      the brackets!)  The text should be enclosed in the appropriate
      comment syntax for the file format. We also recommend that a
      file or class name and description of purpose be included on the
      same "printed page" as the copyright notice for easier
      identification within third-party archives.

   Copyright [yyyy] [name of copyright owner]
```

Fixed in this PR (I know it's out of scope; feel free to cherry-pick the commit).

Pandapip1 commented:
Never mind, I'm stupid. It means to apply it to the source files, not the LICENSE itself.

Pandapip1 commented on CMakeLists.txt:
```diff
@@ -23,6 +23,7 @@ include("CMake/AddDependencies.cmake")
 include("CMake/Fixups.cmake")
 include("CMake/GenerateBindings.cmake")
 include("CMake/GroupSources.cmake")
+include("CMake/SetVersion.cmake")
```
I believe my contribution to this file does not meet the creativity requirement to receive copyright.
